### PR TITLE
fix(jaxopkeys): route TE-adjacent buffer-init kernels to TE

### LINKED ISSUE #422: '__amd_rocclr_fillBufferAligned.kd' events
For events with name '__amd_rocclr_fillBufferAligned.kd', their 'gpu_kernel_op_cat' is 'Uncategorized Events/XLA'. However, some of them do have valid operands shapes and custom_call_target, which indicate that they are TE based fused attention kernels.

### LINKED ISSUE #423: 'FillBuffer' events
Using TraceEventUtils.JaxOpKeys.ClassCategories.keys, the 'gpu_kernel_op_cat' of these events are currenlty classified as Conv.  A closer look at their operands and custom_call_target in the 'metadata' show that they are 'te_fused_attn' or TE events.

### PR DESCRIPTION
**Pull request by gphuang**
## Summary

Closes #422
Also closes #423

JAX events whose **name** doesn't match any `JaxOpKeys` substring (e.g. `__amd_rocclr_fillBufferAligned.kd`) but whose **`args["hlo_op"]`** does (e.g. `te_fused_attn_backward_ffi.12`) were routed to `Uncategorized Events`. Now categorized as `TE` via a metadata fallback.

### Changes

- `JaxAnalyses.breakdown_compute_events`: when the name keyword scan misses, retry against `args["hlo_op"]` before bucketing as uncategorized.
- `JaxOpKeys.TEKeys`: add `"te_fused_attn"` so `te_fused_attn_{forward,backward}_ffi` host events match by name.
- `JaxOpKeys.ConvKeys`: drop `"FillBuffer"` (closes #423 — XLA emits FillBuffer fusions inside TE custom calls; not a real Conv signal).

## Test plan

- [x] 5 unit tests in `tests/test_jax_te_fused_attn_categorization.py` against a 200 KB JAX 0.8 / TE 2.8 / ROCm 7.1.1 fixture (multi-layer fused-attn fwd+bwd, causal mask, dropout)
- [x] `black` clean
- [x] Full JAX test suite passes (24 tests, 2 skipped). 3 unrelated failures pre-exist on `main` (`test_collective_analysis`, `test_roofline_bound::test_origami_time_in_unified_perf_summary`, `test_subtree_kernel_time::...miopen_convolution...`)

### CODE REVIEW

File: TraceLens/TreePerf/jax_analyses.py [hunk 1]
```diff
@@ -63,6 +63,24 @@ def add_event(cur_event_list, name, duration):
...
                     break
+            if not found:
+                # Metadata-aware fallback (issue #422): an XLA-emitted
+                # buffer-init kernel such as `__amd_rocclr_fillBufferAligned.kd`
+                # does not match any name keyword, but its
+                # ``args["hlo_op"]`` carries the surrounding XLA op (e.g.
+                # ``te_fused_attn_backward_ffi.12``) which does. Re-run the
+                # same substring match against that field so the kernel
+                # lands in the correct category instead of `Uncategorized`.
+                hlo_op = (compute_event.get("args") or {}).get("hlo_op")
```
[the remark this answers is not in the document]
Comment (gphuang) [reply TraceLens/TreePerf/jax_analyses.py]: Done in 39fdb88: now uses `TraceEventUtils.TraceKeys.Args` and `TraceEventUtils.JaxKernelEventArgs.hlo_op`.

File: TraceLens/TreePerf/jax_analyses.py [hunk 2]
```diff
@@ -63,6 +63,24 @@ def add_event(cur_event_list, name, duration):
...
+                if isinstance(hlo_op, str) and hlo_op:
+                    for (
+                        category,
+                        filters,
+                    ) in TraceEventUtils.JaxOpKeys.ClassCategories.items():
+                        if any(f in hlo_op for f in filters):
+                            add_event(cur_categorized_list, category, duration)
+                            found = True
+                            break
```
[the remark this answers is not in the document]
Comment (gphuang) [reply TraceLens/TreePerf/jax_analyses.py]: Done in 39fdb88: factored out `_match_category(candidate)` and call it for both `name` and `args["hlo_op"]`. Future key/category changes only touch one place.

File: TraceLens/TreePerf/jax_analyses.py [hunk 3]
```diff
@@ -63,6 +63,24 @@ def add_event(cur_event_list, name, duration):
...
             if not found:
```
[the remark this answers is not in the document]
Comment (gphuang) [reply TraceLens/TreePerf/jax_analyses.py]: Good catch. Done in 39fdb88: `cur_categorized_list.setdefault(UncategorizedEventKey, [0, 0])` runs once per (per-GPU) loop iteration, so `categorized_times[UncategorizedEventKey][1]` in `create_gpu_summary` (jax_analyses.py:171) is now safe. Added `test_uncategorized_bucket_is_always_seeded` to pin the behaviour.

File: tests/test_jax_te_fused_attn_categorization.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  TE-adjacent buffer-init fusion to ``Conv``.
+
+The fixture is a small JAX 0.8 / TE 2.8 / ROCm 7.1.1 trace generated
+by ``learning/jax-minimal/12_te_fused_attn.py`` (2-layer training
+forward + ``jax.value_and_grad``, BS3HD, causal mask, dropout 0.1).
+"""
+
+import os
+
+from TraceLens.TreePerf.gpu_event_analyser import JaxGPUEventAnalyser
```
[the remark this answers is not in the document]
Comment (gphuang) [reply tests/test_jax_te_fused_attn_categorization.py]: Done in 39fdb88.

File: tests/test_jax_te_fused_attn_categorization.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    data = DataLoader.load_data(filename_path=_trace_path(), save_preprocessed=False)
+    events = data["traceEvents"]
+    # GPU-stream compute events: pid<=100, tid<100, with a duration.
+    compute = [
+        e
+        for e in events
+        if isinstance(e.get("name"), str)
+        and "dur" in e
+        and e.get("pid") is not None
+        and int(e.get("pid")) <= 100
```
Comment (gabeweisz) [inline tests/test_jax_te_fused_attn_categorization.py]: I don't like having 100 hard coded here - I know that this does exist elsewhere but we should start to fix it.

Can you put (e.get("pid") is not None
        and int(e.get("pid")) <= 100) into a helper function into jax_analyses? Same for tid

**Review by gabeweisz — APPROVED** [review]

File: tests/test_jax_te_fused_attn_categorization.py (hunk 5 above)
Comment (gphuang) [reply tests/test_jax_te_fused_attn_categorization.py]: ok, fixed it here first.
